### PR TITLE
Add connector table and section for reports

### PR DESCRIPTION
**Pull request by sp35**
Related to #101 

- [x] Add connectors table
- [x] Add Connector service
- [x] Add connector reports section In job analysis page
- [x] Handle connector reports using results service

### CODE REVIEW

File: src/app/app-routing.module.ts [hunk 1]
```diff
@@ -21,6 +21,7 @@ export const routes: Routes = [
 const config: ExtraOptions = {
   useHash: false,
   relativeLinkResolution: 'legacy',
+  enableTracing: true,
```
Comment (Eshaan7) [inline src/app/app-routing.module.ts:24]: what does this do ?

File: src/app/pages/connectors/connectors-calls/connectors-calls.component.html [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<!-- Table -->
+
+<div class="row">
+  <div class="col-sm-12">
+    <nb-card fullWidth>
+      <nb-card-header>
+        <span>Jobs - count: 3</span>
```
Comment (Eshaan7) [inline src/app/pages/connectors/connectors-calls/connectors-calls.component.html:7]: TODO: remove constant value

File: src/app/app-routing.module.ts (hunk 1 above)
Comment (sp35) [reply src/app/app-routing.module.ts:24]: It's for debugging purpose, will remove that later!

File: src/app/@core/models/models.ts [hunk 3]
```diff
@@ -83,6 +84,24 @@ export interface IAnalyzerConfig {
   observable_supported?: string[];
 }
 
+export interface IRawConnectorConfig {
+  [name: string]: IConnectorConfig;
+}
+
+export interface IConnectorConfig {
```
Comment (Eshaan7) [inline src/app/@core/models/models.ts:91]: We should have 2 interfaces. An `IAbstractConfig` interface which could be `extended` into 2: `IConnectorConfig` and `AnalyzerConfig`.

File: src/app/pages/connectors/connectors-table/connectors-table.component.ts [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+          return r.verification.missing_secrets;
+        },
+        renderComponent: JSONRenderComponent,
+      },
+      configured: {
+        title: 'Configured',
+        filter: false,
+        type: 'custom',
+        valuePrepareFunction: (c, r) => {
+          return {
```
Comment (Eshaan7) [inline src/app/pages/connectors/connectors-table/connectors-table.component.ts:66]: No need of extra `return` and curly brackets, would be cleaner if done as an inline arrow function.

File: src/app/pages/dashboard/dashboard.module.ts [hunk 5]
```diff
@@ -40,11 +41,12 @@ import * as echarts from 'echarts';
     DashboardComponent,
     JobResultComponent,
     JobActionsMenuComponent,
-    ViewResultButtonComponent,
+    ViewJobAnalyzersResultButtonComponent,
```
Comment (Eshaan7) [inline src/app/pages/dashboard/dashboard.module.ts:44]: The renaming is pretty redundant now so I think it's best to revert it back to it's original name.

**Comment by Eshaan7** [comment]
https://github.com/intelowlproject/IntelOwl-ng/commit/b979619a700f31c175f107213242a0bdeefb4038
Actually this is a very hacky thing I made to make it work and then forgot about it. 

We should do something like:
- have a property IsLoading; set to `false` by default and set to `true` when the response comes.
- show a spinner until it loads so the template doesn't break

If you are comfortable with rxjs, you can give this a try otherwise I can do it later when I get the time.

File: src/app/pages/dashboard/dashboard.component.ts [hunk 6]
```diff
@@ -100,6 +100,16 @@ export class DashboardComponent implements OnDestroy {
         filter: false,
         renderComponent: JobStatusIconRenderComponent,
       },
+      no_of_connectors_executed: {
```
Comment (Eshaan7) [inline src/app/pages/dashboard/dashboard.component.ts:103]: Have you added a `SerializerMethodField` in the `JobListSerializer` for this ? similar to [this](https://github.com/intelowlproject/IntelOwl/blob/8f14384c315613567d39b3c5aa612d61f05c3b91/api_app/serializers.py#L66)

**Review by Eshaan7 — APPROVED** [review]

File: src/app/pages/dashboard/dashboard.component.ts (hunk 6 above)
Comment (Eshaan7) [reply src/app/pages/dashboard/dashboard.component.ts:103]: https://github.com/orgs/intelowlproject/projects/2#card-64203186

**Pull request merged**